src/coreclr/jit/gentree.cpp

```diff
@@ -19681,6 +19691,11 @@ GenTree* Compiler::gtNewSimdBinOpNode(
                         assert(compIsaSupportedDebugOnly(InstructionSet_AVX2));
                         intrinsic = NI_AVX2_MultiplyLow;
                     }
+                    else if (simdSize == 64)
+                    {
+                        assert(compIsaSupportedDebugOnly(InstructionSet_AVX512F));
+                        intrinsic = NI_AVX512F_MultiplyLow;
+                    }
```

Comment:
You'll also need to handle `TYP_LONG` and `TYP_ULONG` just below

It should pick `NI_AVX512DQ_MultiplyLow` for `simdSize == 64`. It's probably going to assert right now.